src/test/java/org/opensearch/ad/util/ParseUtilsTests.java

```diff
@@ -287,4 +288,29 @@ public void testGetFeatureFieldNames() throws IOException {
         assertTrue(fieldNames.contains("field-name1"));
         assertTrue(fieldNames.contains("field-name2"));
     }
+
+    public void testIsAdmin() {
+        User user1 = new User(
+            randomAlphaOfLength(5),
+            ImmutableList.of(),
+            ImmutableList.of("all_access"),
+            ImmutableList.of(randomAlphaOfLength(5))
+        );
+        assertTrue(isAdmin(user1));
+    }
+
+    public void testIsAdminBackendRoleIsAllAccess() {
+        String backendRole1 = "all_access";
+        User user1 = new User(
+            randomAlphaOfLength(5),
+            ImmutableList.of(backendRole1),
+            ImmutableList.of(randomAlphaOfLength(5)),
+            ImmutableList.of(randomAlphaOfLength(5))
+        );
+        assertFalse(isAdmin(user1));
```

Comment:
curiosity question: a bit confused on this test case. If we're just checking if the role contains `all_access` keyword to decide if the role is admin, why would this test case fail? Cause in this test case, the backendRole1 seems also contains `all_access`?

> `user.getRoles().contains("all_access")`

Reply:
This test case is failing because all_access is added as a backend role and not as a role. This is to make sure user can't just be given backend role name of `all_access` and bypass our `isAdmin()` check which is only if they actually have the role itself of `all_access`. Its a little confusing terminology but backend_role are more  Arbitrary strings that you can give to a user to either easily map roles or to identify the group they are in so for example we can use it to filter which detectors they can see or not.